webhooks/cluster_test.go

```diff
@@ -96,71 +91,43 @@ func TestClusterValidation(t *testing.T) {
 		{
 			name:      "should return error when cluster namespace and infrastructure ref namespace mismatch",
 			expectErr: true,
-			in: &clusterv1.Cluster{
-				ObjectMeta: metav1.ObjectMeta{
-					Namespace: "foo",
-				},
-				Spec: clusterv1.ClusterSpec{
-					InfrastructureRef: &corev1.ObjectReference{
-						Namespace: "bar",
-					},
-					ControlPlaneRef: &corev1.ObjectReference{
-						Namespace: "foo",
-					},
-				},
-			},
+			in: builder.Cluster("foo", "cluster1").
```

Comment:
To me the old version is easier to instantly grasp as it's 100% obvious what the namespaces are. That isn't the case with the builder.
(only applies to those 4 cases here, the others are fine as usually you have something like `WithVersion` which helps)

Reply:
WDYT about calling the namespaces `fooNamespace` and `barNamespace` instead of `foo`/`bar` to make it more obvious?